src/sentry/services/hybrid_cloud/organization/impl.py

```diff
@@ -92,6 +92,56 @@ def check_membership_by_email(
 
         return serialize_member(member)
 
+    def check_membership_by_invite_token(
+        self, organization_id: int, org_member_id: int, invite_token: str
+    ) -> Optional[RpcOrganizationMember]:
+        try:
+            member = OrganizationMember.objects.get(
+                id=org_member_id, organization_id=organization_id, token=invite_token
+            )
+        except OrganizationMember.DoesNotExist:
+            return None
+
+        return serialize_member(member)
+
+    def get_organization_member(
+        self, *, organization_member_id: int
+    ) -> Optional[RpcOrganizationMember]:
+        try:
+            member = OrganizationMember.objects.get(id=organization_member_id)
+        except OrganizationMember.DoesNotExist:
+            return None
+
+        return serialize_member(member)
+
+    def delete_organization_member(self, *, organization_member_id: int) -> bool:
+        try:
+            member = OrganizationMember.objects.get(id=organization_member_id)
+        except OrganizationMember.DoesNotExist:
+            return False
+        num_deleted, _deleted = member.delete()
+        return num_deleted > 0  # type: ignore[no-any-return]
+
+    def set_user_for_organization_member(
+        self,
+        *,
+        organization_member_id: int,
+        user_id: int,
+    ) -> Optional[RpcOrganizationMember]:
+        region_outbox = None
+        with transaction.atomic():
+            try:
+                org_member = OrganizationMember.objects.get(id=organization_member_id)
+                org_member.set_user(user_id)
+                org_member.save()
+                region_outbox = org_member.outbox_for_update()
+                region_outbox.save()
```

Comment:
We do this pattern quite a bit, should we have a method on the member that does a save + outbox? Perhaps `save_with_outbox()` :shrug: